sessions/forecasting-nowcasting.qmd

```diff
@@ -0,0 +1,798 @@
+- Forecasts are then drawn from the predictive distribution for each future day, ensuring that both nowcasting and forecasting are handled within a single, consistent generative framework.
+- This is exactly the same as for the `renewal_model` as we are only forecasting the final number of onsets and not simulating the reporting process.
+
+:::
+
+Now we fit the joint model with forecasting parameters:
+
+```{r joint-fit}
+joint_forecast_fit <- nfidd_sample(
+  joint_model,
+  data = list(
+    n = length(unique(filtered_reporting_triangle$day)),
+    m = nrow(filtered_reporting_triangle),
+    p = filtered_reporting_triangle |>
+     group_by(day) |>
+     filter(d == max(d)) |>
+     mutate(d = d + 1) |>
+     pull(d),
+    obs = filtered_reporting_triangle$reported_onsets,
+    d = 16,
+    gen_time_max = length(gen_time_pmf),
+    gen_time_pmf = gen_time_pmf,
+    ip_max = length(ip_pmf) - 1,
+    ip_pmf = ip_pmf,
+    h = forecast_horizon
+  ),
+  adapt_delta = 0.99,
+  max_treedepth = 12
+)
+```
+
+We extract the joint results for forecasting:
+
+```{r joint-forecast-extract}
+joint_forecasts <- joint_forecast_fit |>
+  gather_draws(forecast[day]) |>
+  ungroup() |>
+  filter(.draw %in% sample(.draw, 100)) |>
+  mutate(
+    actual_day = day + max(filtered_reporting_triangle$day),
+    approach = "Joint"
+  )
+
+tail(joint_forecasts)
+```
+
+Finally, let's visualise the joint nowcast results:
+
+```{r plot-joint-nowcast}
+# Plot joint forecast from cutoff onwards
+forecast_start_day <- cutoff
+forecast_end_day <- cutoff + forecast_horizon
+
+ggplot() +
+  # Final observed data (truth) for evaluation period
+  geom_point(data = complete_at_horizon |> 
+               filter(day >= forecast_start_day, day <= forecast_end_day), 
+            aes(x = day, y = complete_onsets), 
+            color = "black", size = 1) +
+  # Joint model forecast ribbons from cutoff
+  geom_line(data = joint_forecasts |>
+              filter(actual_day >= forecast_start_day, actual_day <= forecast_end_day),
+            aes(x = actual_day, y = .value, group = .draw),
+            color = "purple", alpha = 0.1) +
+  # Mark cutoff date
+  geom_vline(xintercept = cutoff, 
+             linetype = "dashed", color = "blue", linewidth = 1) +
+  scale_y_log10() +
+  xlim(forecast_start_day, forecast_end_day) +
+  labs(
+    x = "Day",
+    y = "Onsets",
+    title = "Joint approach: Forecasting from joint nowcast and forecast",
+    subtitle = "Blue line: cutoff (forecast start)"
+  ) +
+  theme_minimal()
+```
+
+::: {.callout-tip}
+## Take 5 minutes
+
+What are the advantages of the joint model approach?
+- How does it address the model mismatch problem?
+- What uncertainties does it properly account for?
+- Why might it perform better than the pipeline and complete data approaches?
+- Why might it perform worse than the complete data approach?
+
+:::
+
+::: {.callout-note collapse="true"}
+## Solution
```

Comment:
The formatting for these answers is a little bit weird when rendered. Like a lot of separate non-bulleted sentences.